### PR TITLE
Apply security best practices

### PR DESCRIPTION
**Pull request by mmorel-35**
## What does this PR do?

Apply security best practices

## Why is it important?

Improve security of the project

### CODE REVIEW

**Comment by mdelapenya** [comment]
@mmorel-35 could you please elaborate on the [harden-runner action](https://github.com/step-security/harden-runner)? I'm concerned about its usage, although I can see it's used by very important agencies (like CISA)

**Comment by mmorel-35** [comment]
OK, let’s get rid of it and stick to permissions and hashed versions then.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -143,17 +143,17 @@ jobs:
...
           fetch-depth: 0
       
-      - uses: actions/download-artifact@v3
+      - uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
         with:
           name: sonarcloud
             
       - name: Analyze with SonarCloud
-        uses: sonarsource/sonarcloud-github-action@master
+        uses: sonarsource/sonarcloud-github-action@5ee47de3c96f0c1c51b09d2ff1fec0cfeefcf67c # master
```
Comment (mmorel-35) [inline .github/workflows/ci.yml:156]:
```suggestion
        uses: sonarsource/sonarcloud-github-action@master
```

File: modulegen/_template/ci.yml.tmpl [hunk 2]
```diff
@@ -143,17 +143,17 @@ jobs:
...
           fetch-depth: 0
       
-      - uses: actions/download-artifact@v3
+      - uses: actions/download-artifact@9bc31d5ccc31df68ecc42ccf4149144866c47d8a # v3.0.2
         with:
           name: sonarcloud
             
       - name: Analyze with SonarCloud
-        uses: sonarsource/sonarcloud-github-action@master
+        uses: sonarsource/sonarcloud-github-action@5ee47de3c96f0c1c51b09d2ff1fec0cfeefcf67c # master
```
Comment (mmorel-35) [inline modulegen/_template/ci.yml.tmpl:156]:
```suggestion
        uses: sonarsource/sonarcloud-github-action@master
```

**Pull request closed without merging**